stdlib.go

```diff
@@ -0,0 +1,66 @@
+package argot
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/kylelemons/godebug/pretty"
+)
+
+// ExpectNil is a Step that checks if the value is nil.
+func ExpectNil(actual interface{}) *NamedStep {
+	return NewNamedStep("ExpectNil", func() error {
+		if actual == nil {
+			return nil
+		}
+		return fmt.Errorf("Expected %#v, got %#v", nil, actual)
+	})
+}
+
+// ExpectDeepEqual is a Step that checks if the actual and expected values are
+// the same, as determined by reflect.DeepEqual.  The error message will
+// mention both the Go representation as well as the type of both values.
+func ExpectDeepEqual(actual interface{}, expected interface{}) *NamedStep {
+	return NewNamedStep("ExpectDeepEqual", func() error {
+		if !reflect.DeepEqual(actual, expected) {
+			return fmt.Errorf("Expected %#[1]v (%[1]T), got %#[2]v (%[2]T)", actual, expected)
```

Comment:
surely the indices aren't needed?
And don't you have the args reversed here?

Reply:
Indices where so the arguments don't have to be specified twice, anyway, removed that. Reversed, yes, indeed.